corehq/apps/translations/app_translations/upload_form.py

```diff
@@ -158,13 +158,13 @@ def _add_or_remove_translations(self, lang, row):
         label_id = row['label']
         translations = self._get_translations_for_row(row, lang)
         translation_node = self.itext.find("./{f}translation[@lang='%s']" % lang)
-        keep_value_node = any(v for k, v in translations.items())
+        keep_value_node = any(v for k, v in translations.items()) or set(self.langs) != set(self.app.langs)
         text_node = self._get_text_node(translation_node, label_id)
         for trans_type, new_translation in translations.items():
-            if not new_translation:
+            if not new_translation and len(self.langs) > 1:
```

Comment:
If there is actually just one lang in the app this block would never run?

Reply:
Yes - if there's one language and the translation is blank, there aren't any other languages to fall back to, so this block is irrelevant.